### PR TITLE
Respect min_tokens_to_keep in TopHLogitsWarper

### PR DESCRIPTION
**Pull request by Incheonkirin**
# What does this PR do?

`generate()` computes a `min_tokens_to_keep` value for sampling, using a larger value for beam methods so at least one non-EOS continuation can survive. That value is already passed to the top-k, top-p, min-p, typical, epsilon, and eta warpers, but `TopHLogitsWarper` was constructed without it.

As a result, with peaked logits and beam sampling, top-h can keep only the single highest-probability token while the other sampling warpers keep the beam-safe minimum. This PR adds `min_tokens_to_keep` to `TopHLogitsWarper` and passes the generation-level value through, matching the behavior of the other truncation samplers.

The default remains `1`, so existing non-beam top-h behavior is unchanged.

## Tests

Added a regression case to `test_top_h_dist_warper` that checks `TopHLogitsWarper(top_h=0.3, min_tokens_to_keep=2)` keeps two tokens on a highly peaked distribution.

Local:

```bash
PYTHONPATH=src python -m pytest tests/generation/test_logits_process.py -q
# 44 passed
```

## Code Agent Policy

- [X] I confirm that this is not a pure code agent PR.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

### CODE REVIEW

**Comment by Incheonkirin** [comment]
Replacement PR is now #47262 from a proper fork. It contains the rebased change and current validation; no reopen is needed.
